Sources/NetworkService/NetworkError.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+
+/// Contains network errors that may occur, while the application is running.
+public enum NetworkError: Error, Codable, CaseIterable {
```

Comment:
why is Codable and CaseIterable here?

Reply:
I've removed `Codable` protocol. You are right, it's unneccesary here. But we need `CaseIterable` protocol to have opportunity to test enum's cases.